test/e2e/specs/editor/various/preview.spec.js

```diff
@@ -0,0 +1,342 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.use( {
+	previewUtils: async ( { page }, use ) => {
+		await use( new PreviewUtils( { page } ) );
+	},
+} );
+
+test.describe( 'Preview', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'should open a preview window for a new post', async ( {
+		editor,
+		page,
+		pageUtils,
+		previewUtils,
+		requestUtils,
+	} ) => {
+		const editorPage = page;
+
+		// Disabled until content present.
+		await expect(
+			editorPage.locator( 'role=button[name="Preview"i]' )
+		).toBeDisabled();
+
+		await editorPage.type(
+			'role=textbox[name="Add title"]',
+			'Hello World'
+		);
+
+		const previewPage = await editor.openPreviewPage( editorPage );
+		const previewTitle = previewPage.locator( 'role=heading[level=1]' );
+
+		await expect( previewTitle ).toHaveText( 'Hello World' );
+
+		// When autosave completes for a new post, the URL of the editor should
+		// update to include the ID. Use this to assert on preview URL.
+		await expect( editorPage ).toHaveURL( /[\?&]post=(\d+)/ );
+		const [ , postId ] = editorPage.url().match( /[\?&]post=(\d+)/ );
+
+		const expectedPreviewURL = new URL( requestUtils.baseURL );
+		expectedPreviewURL.search = `?p=${ postId }&preview=true`;
+		await expect( previewPage ).toHaveURL( expectedPreviewURL.href );
+
+		// Return to editor to change title.
+		await editorPage.bringToFront();
+		await editorPage.type( 'role=textbox[name="Add title"]', '!' );
+		await previewUtils.waitForPreviewNavigation( previewPage );
+
+		// Title in preview should match updated input.
+		await expect( previewTitle ).toHaveText( 'Hello World!' );
+
+		// Pressing preview without changes should bring same preview window to
+		// front and reload, but should not show interstitial.
+		await editorPage.bringToFront();
+		await previewUtils.waitForPreviewNavigation( previewPage );
+
+		await expect( previewTitle ).toHaveText( 'Hello World!' );
+
+		// Preview for published post (no unsaved changes) directs to canonical URL for post.
+		await editorPage.bringToFront();
+		await editor.publishPost();
+
+		// Close the panel.
+		await page.click( 'role=button[name="Close panel"i]' );
+
+		// Return to editor to change title.
+		await editorPage.bringToFront();
+		await editorPage.click( 'role=textbox[name="Add title"i]' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'A' );
+		await editorPage.keyboard.press( 'ArrowRight' );
+		await editorPage.keyboard.type( ' And more.' );
```

Comment:
Let's change this to `.fill()` to simplify it a bit:

```suggestion
		await editorPage.fill(
			'role=textbox[name="Add title"i]',
			'Hello World! And more.'
		);
```